File: euler_3.py
def primefactors(num):
  startnum=2
  primeNumbers=list()
  
  quota=0
  rest=0
  
  while True:
    quota=num/startnum
    rest=num%startnum
    
    #print("quota="+quota+", rest="+rest+", startnum="+startnum)
      
    if rest==0:
      primeNumbers.append(startnum)
      startnum=2
      num=quota
    else:
      startnum=startnum+1
      
    if quota==1 and rest==0:
      break
  
  return primeNumbers

primeNumbers=primefactors(600851475143)
print (primeNumbers)

primeNumbers.sort(reverse=True)
print (primeNumbers[0])

File: need4spd/euler_3.rb
def primefactors(num)
  startnum=2
  primeNumbers=Array.new
  
  quota=0
  rest=0
  
  while true
    quota=num/startnum
    rest=num%startnum
    puts "quota=#{quota}, rest=#{rest}, startnum=#{startnum}"
      
    if rest==0
      primeNumbers.push(startnum)
      startnum=2
      num=quota
    else
      startnum=startnum+1
    end
      
    if quota==1 and rest==0
      break
    end
    
  end
  
  return primeNumbers
end

primeNumbers=primefactors(600851475143)
puts primeNumbers

sortedList=primeNumbers.sort {|x,y| y<=>x}
puts sortedList[0]

